### DISCUSSION TITLE
How do you use gdb for debugging?

### DISCUSSION THREAD

**Discussion by ConstStar** [nasa/fprime#2356] [Q&A] [answered]
And how are debugging use in cross-compilation.
Thanks.

**Comment by VR314** [comment, marked as answer]
In your CMakeFiles.txt for your deployment, include the following lines (I included them below the `cmake_minimum_required` line:
```
set(CMAKE_BUILD_TYPE Debug)
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
```
After building, run `gdb ./path/to/executable` and it should work.

**Reply by ConstStar** [reply]
ok,thanks.
